src/instruction/function_declaration.rs

```diff
@@ -187,6 +190,33 @@ impl Instruction for FunctionDec {
     }
 }
 
+impl TypeCheck for FunctionDec {
+    fn resolve_type(&self, ctx: &mut Context) -> CheckedType {
+        let expected_ty = match &self.ty {
+            // FIXME: Remove clone?
+            Some(ty) => CheckedType::Resolved(ty.clone()),
```

Comment:
shouldn't this be ty.resolve_type() ?

Reply:
The `ty` field is simply a `TypeId`, which is not an instruction so it cannot resolve to a type. It already is a type